extensions/metadata/interface/src/main/java/org/exist/storage/md/MDStorageManager.java

```diff
@@ -41,291 +43,339 @@
  *
  */
 public class MDStorageManager implements Plug, BackupHandler, RestoreHandler {
-	
+
     protected final static Logger LOG = LogManager.getLogger(MDStorageManager.class);
 
-    public final static String PREFIX = "md";
-	public final static String NAMESPACE_URI = "http://exist-db.org/metadata";
-
-	public final static String UUID = "uuid";
-	public final static String META = "meta";
-	public final static String KEY = "key";
-	public final static String VALUE = "value";
-	public final static String VALUE_IS_DOCUMENT = "value-is-document";
-	
-	public final static String PREFIX_UUID = PREFIX+":"+UUID;
-	public final static String PREFIX_KEY = PREFIX+":"+KEY;
-	public final static String PREFIX_META = PREFIX+":"+META;
-	public final static String PREFIX_VALUE = PREFIX+":"+VALUE;
-	public final static String PREFIX_VALUE_IS_DOCUMENT = PREFIX+":"+VALUE_IS_DOCUMENT;
-
-	protected static MDStorageManager inst = null;
-	
-	MetaData md;
-	
-	public MDStorageManager(PluginsManager manager) throws PermissionDeniedException {
-		try {
-			@SuppressWarnings("unchecked")
-			Class<? extends MetaData> backend = 
-				(Class<? extends MetaData>) Class.forName("org.exist.storage.md.MetaDataImpl");
-		
-			Constructor<? extends MetaData> ctor = backend.getConstructor(Database.class);
-			md = ctor.newInstance(manager.getDatabase());
-		} catch (Exception e) {
-			e.printStackTrace();
-			throw new PermissionDeniedException(e);
-		}
-
-		inst = this;
-		
-		Database db = manager.getDatabase();
-		
-		inject(db, md);
+//    public final static String LUCENE_ID = "org.exist.indexing.lucene.LuceneIndex";
+
+    public final static String UUID = "uuid";
+    public final static String META = "meta";
+    public final static String KEY = "key";
+    public final static String VALUE = "value";
+    public final static String VALUE_IS_DOCUMENT = "value-is-document";
+
+    public final static String PREFIX_UUID = PREFIX+":"+UUID;
+    public final static String PREFIX_KEY = PREFIX+":"+KEY;
+    public final static String PREFIX_META = PREFIX+":"+META;
+    public final static String PREFIX_VALUE = PREFIX+":"+VALUE;
+    public final static String PREFIX_VALUE_IS_DOCUMENT = PREFIX+":"+VALUE_IS_DOCUMENT;
+
+    protected static MDStorageManager inst = null;
+
+    protected static MDStorageManager get() {
+        return inst;
+    }
+
+    MetaData md;
+
+    public MDStorageManager(PluginsManager manager) throws PermissionDeniedException {
+        try {
+            @SuppressWarnings("unchecked")
+            Class<? extends MetaData> backend =
+                (Class<? extends MetaData>) Class.forName("org.exist.storage.md.MetaDataImpl");
+
+            Constructor<? extends MetaData> ctor = backend.getConstructor(Database.class);
+            md = ctor.newInstance(manager.getDatabase());
+        } catch (Exception e) {
+                e.printStackTrace();
+            throw new PermissionDeniedException(e);
+        }
+
+        inst = this;
```

Comment:
The instantiation of this object is very strange! It looks to me like you want it to be a singleton? If so, can you please switch it to a true singleton patten, mixing a static non-final instance and then exposing a public constructor is not a good choice.